packages/nx-aws-cache/src/tasks-runner/aws-cache.ts

```diff
@@ -126,9 +128,15 @@ export class AwsCache implements RemoteCache {
   private async createAndUploadFile(hash: string, cacheDirectory: string): Promise<boolean> {
     try {
       const tgzFilePath = this.getTgzFilePath(hash, cacheDirectory);
-
       await this.createTgzFile(tgzFilePath, hash, cacheDirectory);
-      await this.uploadFile(hash, tgzFilePath);
+      const sourceFileStream = createReadStream(tgzFilePath);
+
+      await this.uploadFile(
+        hash,
+        this.encryptConfig
+          ? sourceFileStream.pipe(new Encrypt(this.encryptConfig))
```

Comment:
I just noticed a problem here. Why would you make a class instance each time when uploading a file? This can bo done once in `AwsCache` class constructor and reuse it.

Reply:
I move both Stream Transform to constructor

Reply:
I moved back create class instance when upload a file.
If you will create Singleton, readFirstChunk property in Decrypt and Encrypt classes always be true and encrypted files will be damaged.

Reply:
I see, ok.